### PR TITLE
[#128] Make case comparisons consistent between various locale settin…

### PR DESCRIPTION
**Pull request by monkpow**
…g methods (e.g. cookies, headers, query).

### CODE REVIEW

**Comment by monkpow** [comment]
In the existing code line, the behavior is a bit different when using queryParameters to set locale in the case of a language code like en-US.

if setting via:
   cookie => en-US
   header => en-US
   queryParameter => en-us

After reading it a few times, I suspect this is a bug, but have wrapped it in a config value just in case some current users rely on this behavior.

File: test/i18n.configureCookiename.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var i18n = require('../i18n'),
   should = require("should"),
   path = require("path");
 
-describe('Locale switching should work queryParameter', function() {
+describe('Locale switching should work when set via cookie', function() {
```
Comment (monkpow) [inline test/i18n.configureCookiename.js]: Not really related to this change set, but seemed polite to clean this up.

**Comment by mashpie** [comment]
Thank you sir! It'd be best to handle those consistent internally. Refactoring such is still on my backlog, so your help catching up is much appreciated. I will review some extra changes on weekend to merge an publish.

**Comment by monkpow** [comment]
Ok thanks, let me know if you'd prefer I pull out the config switch, and just fix it as a bug.

File: test/i18n.configureCookiename.js (hunk 1 above)
Comment (mashpie) [inline test/i18n.configureCookiename.js]: yap I've seen this... thanks again - I bet there are still some more typos in readme, comments & tests - 6 years of part time maintenance...

**Comment by mashpie** [comment]
the switch is fine - I am a big fan of feature switches and legacy switches could help with issues until next major release.

**Pull request merged**